### PR TITLE
ddebug submodule update

### PR DESCRIPTION
**Pull request by winterheart**
## Pull Request Type

- [ ] GitHub Workflow changes
- [ ] Documentation or Wiki changes
- [x] Build and Dependency changes
- [ ] Runtime changes
  - [ ] Render changes
  - [ ] Audio changes
  - [ ] Input changes
  - [ ] Network changes
  - [x] Other changes

### Description

This PR refactors and unifies ddebug submodule to use `SDL_assert()` for all platforms. Since `SDL_assert()` implements own `debug_break` functionality, bundled version is redundant. Synced DMFC macros with ddebug since there was some compatibility issues. Error handling (`pserror.h`) is also moved from misc submodule to ddebug.

### Screenshots (if applicable)

SDL_assert window on Linux
SDL_assert

SDL_assert window on Windows
Descent-new-debug-break

### Checklist

- [x] I have tested my changes locally and verified that they work as intended.
- [x] I have documented any new or modified functionality.
- [x] I have reviewed the changes to ensure they do not introduce any unnecessary complexity or duplicate code.
- [x] I understand that by submitting this pull request, I am agreeing to license my contributions under the project's license.

### CODE REVIEW

File: ddebug/pserror.h [hunk 1]
```diff
@@ -161,87 +163,81 @@ void AssertionFailed(const char *expstr, const char *file, int line);
...
     if (DebugBreak_callback_stop)                                                                                      \
       (*DebugBreak_callback_stop)();                                                                                   \
-    debug_break();                                                                                                     \
+    SDL_assert(false);                                                                                                 \
     if (DebugBreak_callback_resume)                                                                                    \
       (*DebugBreak_callback_resume)();                                                                                 \
   } while (0)
 
+// Does a DEBUG_BREAK() if debugging is turned on.
 #define Int3()                                                                                                         \
```
Comment (Lgt2x) [inline ddebug/pserror.h:216]: while we're renaming stuff, we should also rename `Int3` to something like `abort` (you can suggest a better name), Int3 is an assembly instruction without much semantic

File: ddebug/pserror.h [hunk 2]
```diff
@@ -161,87 +163,81 @@ void AssertionFailed(const char *expstr, const char *file, int line);
...
   do {                                                                                                                 \
-    mprintf(0, "Int3 at %s line %d.\n", __FILE__, __LINE__);                                                           \
-    if (Debug_break)                                                                                                   \
-      DEBUG_BREAK();                                                                                                   \
-    else                                                                                                               \
-      Int3MessageBox(__FILE__, __LINE__);                                                                              \
+    LOG_ERROR.printf("Int3 at %s:%d.", __FILE__, __LINE__);                                                            \
```
Comment (Lgt2x) [inline ddebug/pserror.h:218]: Fatal error at...

File: ddebug/pserror.h [hunk 3]
```diff
@@ -161,87 +163,81 @@ void AssertionFailed(const char *expstr, const char *file, int line);
...
     if (_heapchk() != _HEAPOK)                                                                                         \
       Int3();                                                                                                          \
   } while (0)
-#elif defined(POSIX)
 
-#include <SDL.h>
+#elif defined(POSIX)
 
-#define ASSERT(x) SDL_assert(x)
 #define HEAPCHECK()
```
Comment (Lgt2x) [inline ddebug/pserror.h:241]: I don't think this is used

File: ddebug/pserror.h (hunk 1 above)
Comment (winterheart) [reply ddebug/pserror.h:216]: There too much `Int3()` usage, PR would be much wider. If this OK, I'd propose `ABORT()`

Comment (Lgt2x) [reply ddebug/pserror.h:216]: I'm fine with `ABORT`

**Review by Lgt2x — APPROVED** [review]
Looks good, tested on Linux Debug & Release builds

**Pull request merged**